AWSS3/AWSS3TransferUtility.m

```diff
@@ -125,6 +126,9 @@ + (instancetype)S3TransferUtilityForKey:(NSString *)key {
 }
 
 + (void)removeS3TransferUtilityForKey:(NSString *)key {
+    AWSS3TransferUtility *transferUtility = [self S3TransferUtilityForKey:key];
+    if (transferUtility)
+        [transferUtility.session invalidateAndCancel];
     [_serviceClients removeObjectForKey:key];
```

Comment:
I won't have the key in the delegate method, so I'll have to either have to:

1. search the dictionary for the object 
2. extract the key from the `_sessionIdentifier`
3. add the key as a property on `AWSS3TransferUtility` 

Options 2 and 3 are pretty ugly, so I'd go with option 1.  Your thoughts?